src/MainRegistry.sol

```diff
@@ -140,6 +140,46 @@ contract MainRegistry is IMainRegistry, MainRegistryGuardian {
         }
     }
 
+    /**
+     * @notice Sets a new oracle for the rate baseCurrency-USD.
+     * @param baseCurrency The identifier of the baseCurrency for which the new oracle is set.
+     * @param newOracle The new oracle address.
+     * @dev This function is part of an oracle failsafe mechanism.
+     * New oracles can only be set if the current oracle is not performing as intended:
+     * - A call to the oracle reverts
+     * - The oracle returns the minimum value
+     * - The oracle returns the maximum value
+     * - The oracle didn't upadet for over a week
+     * @dev This function could be called to set an oracle address for the basecurrency USD (since it is initiated with the zero address).
+     * This oracle is however never used, hence would not cause any problems (exept gas waste).
```

Comment:
should we consider adding a check 'if not basecurrency == 0' then?

Reply:
Not really, we never use that oracle, so it would not really matter if a bogus oracle is set